internal/controller/sriov_controller.go

```diff
@@ -0,0 +1,297 @@
+	for _, rail := range cfg.SpectrumXNetworks.Rails {
+		network := r.generateOVSNetwork(&rail, cfg.SpectrumXNetworks.MTU)
+		logr.Info("Creating OVSNetwork", "name", network.Name)
+		if err := r.Client.Patch(ctx, network, client.Apply, client.ForceOwnership, client.FieldOwner(spectrumXSRIOVControllerName)); err != nil {
+			return fmt.Errorf("error while patching %s %s: %w", network.GetObjectKind().GroupVersionKind().String(), client.ObjectKeyFromObject(network), err)
+		}
+	}
+	return nil
+}
+
+func (r *SRIOVReconciler) getDeviceNameByRail(deviceMapping []config.RailDeviceMapping, railName string) string {
+	for _, mapping := range deviceMapping {
+		if mapping.RailName == railName {
+			return mapping.DevName
+		}
+	}
+	return ""
+}
+
+// reconcileSRIOVNetworkNodePolicy generates SRIOVNetworkNodePolicy according to defined Rails
+func (r *SRIOVReconciler) reconcileSRIOVNetworkNodePolicy(ctx context.Context, cfg *config.Config) error {
+	logr := log.FromContext(ctx)
+	for _, railDevMaping := range cfg.RailDeviceMapping {
+		devName := r.getDeviceNameByRail(cfg.RailDeviceMapping, railDevMaping.RailName)
+		network := r.generateSRIOVNetworkNodePolicy(railDevMaping.RailName, devName, cfg.SpectrumXNetworks.MTU)
+		logr.Info("Creating OVSNetwork", "name", network.Name)
+		if err := r.Client.Patch(ctx, network, client.Apply, client.ForceOwnership, client.FieldOwner(spectrumXSRIOVControllerName)); err != nil {
+			return fmt.Errorf("error while patching %s %s: %w", network.GetObjectKind().GroupVersionKind().String(), client.ObjectKeyFromObject(network), err)
+		}
+	}
+	return nil
+}
+
+// generateOVSNetwork generates a OVSNetwork object for the given Rail
+func (r *SRIOVReconciler) generateOVSNetwork(rail *config.Rail, mtu uint) *sriovnetworkv1.OVSNetwork {
+	network := &sriovnetworkv1.OVSNetwork{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rail.Name,
+			Namespace: r.OVSNetworkNamespace,
+			Labels: map[string]string{
+				spectrumXSRIOVControllerName: "",
+			},
+		},
+		Spec: sriovnetworkv1.OVSNetworkSpec{
+			NetworkNamespace: ovsNetworkNamespace,
+			ResourceName:     rail.Name,
+			IPAM:             fmt.Sprintf(`{"type": "nv-ipam", "poolName": "%s", "poolType": "cidrpool"}`, rail.Name),
+			MTU:              mtu,
+			InterfaceType:    ovsNetworkInterfaceType,
+		},
+	}
+
+	network.ObjectMeta.ManagedFields = nil
+	network.SetGroupVersionKind(sriovnetworkv1.GroupVersion.WithKind(ovsNetworkType))
+	return network
+}
+
+// generateSRIOVNetworkNodePolicy generates a SRIOVNetworkNodePolicy object for the given Rail
+func (r *SRIOVReconciler) generateSRIOVNetworkNodePolicy(railName, deviceName string, mtu uint) *sriovnetworkv1.SriovNetworkNodePolicy {
+	nicSelector := &sriovnetworkv1.SriovNetworkNicSelector{
+		PfNames: []string{deviceName},
+	}
+	bridge := &sriovnetworkv1.Bridge{
+		OVS: &sriovnetworkv1.OVSConfig{
+			Bridge: sriovnetworkv1.OVSBridgeConfig{
+				DatapathType: "netdev",
+				OtherConfig:  map[string]string{"dpdk-init": "true"},
```

Comment:
Let's remove `OtherConfig` for now, and add it in follow-up

Reply:
that's a part of bridge setup. This functionality is available in sriov operator onw

Reply:
ack

Reply:
i think that for doca there are few other configurations that are needed, who is adding `doca-init="true"` and `hw-offload="true"`

Reply:
In the ansible repo, these settings are on the daemon level not the bridge, AFAIU:

```    ovs-vsctl set Open_vSwitch . other_config:dpdk-extra='-a 0000:00:00.0'
    ovs-vsctl set Open_vSwitch . other_config:dpdk-init=true
    ovs-vsctl set Open_vSwitch . other_config:doca-init=true```